sample_project/Assets/SampleViewer/Samples/Routing/RouteManager.cs

```diff
@@ -148,12 +149,17 @@ void Start()
         lastRootPosition = arcGISMapComponent.GetComponent<HPRoot>().RootUniversePosition;
     }
 
-    /// <summary>
-    /// Return GeoPosition Based on RaycastHit; I.E. Where the user clicked in the Scene.
-    /// </summary>
-    /// <param name="hit"></param>
-    /// <returns></returns>
-    private ArcGISPoint HitToGeoPosition(RaycastHit hit, float yOffset = 0)
+	private void Update()
+	{
+        RebaseRoute();
```

Comment:
```suggestion
        `arcGISMapComponent.RootChanged.AddListener(RebaseRoute);`
```
Map component has an event we can use instead of checking every frame. We could add this line to `Start`, and only call rebase when needed.

Reply:
great feedback!